tests/miral/test_window_manager_tools.cpp

```diff
@@ -0,0 +1,297 @@
+                window_manager_policy = policy.get();
+                window_manager_tools = tools;
+                return policy;
+            }
+    }
+{
+}
+
+mt::TestWindowManagerTools::~TestWindowManagerTools() = default;
+
+auto mt::TestWindowManagerTools::create_surface(
+    std::shared_ptr<mir::scene::Session> const& session,
+    mir::scene::SurfaceCreationParameters const& params) -> mir::frontend::SurfaceId
+{
+    // This type is Mir-internal, I hope we don't need to create it here
+    std::shared_ptr<mir::frontend::EventSink> const sink;
+    return session->create_surface(params, sink);
+}
+
+auto mt::TestWindowManagerTools::create_mock_display_configuration(std::vector<miral::Rectangle> outputs)
+    -> std::shared_ptr<graphics::DisplayConfiguration const>
+{
+    auto const display_config = std::make_shared<const mir::test::doubles::MockDisplayConfiguration>();
+    EXPECT_CALL(*display_config, for_each_output(testing::_))
+        .WillRepeatedly(testing::Invoke([outputs](std::function<void(mir::graphics::DisplayConfigurationOutput const&)> func){
+            for (auto i = 0u; i < outputs.size(); i++)
+            {
+                auto const& rect = outputs[i];
+                mir::graphics::DisplayConfigurationOutput display_config_output{
+                    mir::graphics::DisplayConfigurationOutputId{(int)i}, // id
+                    mir::graphics::DisplayConfigurationCardId{1}, // card_id
+                    mir::graphics::DisplayConfigurationOutputType::unknown, // type
+                    {mir_pixel_format_abgr_8888}, // pixel_formats
+                    {{rect.size, 60}}, // modes
+                    0, // prefered_mode_index
+                    {rect.size}, // physical_size_mm
+                    true, // connected
+                    true, // used
+                    rect.top_left, // top_left
+                    0, // current_mode_index
+                    mir_pixel_format_abgr_8888, // current_format
+                    mir_power_mode_on, // power_mode
+                    mir_orientation_normal, // orientation
+                    1.0, // scale
+                    mir_form_factor_unknown, // form_factor
+                    mir_subpixel_arrangement_unknown, // subpixel_arrangement
+                    {}, // gamma
+                    mir_output_gamma_unsupported, // gamma_supported
+                    {}, // edid
+                    {}, // custom_logical_size
+                };
+                func(display_config_output);
+            }
+        }));
+    return display_config;
+}
+
+void mt::TestWindowManagerTools::notify_configuration_applied(
+    std::shared_ptr<graphics::DisplayConfiguration const> display_config)
+{
+    auto config_observer = self->display_configuration_observer.observer.lock();
+    ASSERT_THAT(config_observer, testing::NotNull());
+    config_observer->configuration_applied(display_config);
```

Comment:
I'm not convinced this logic belongs here and not on FakeDisplayConfigurationObserverRegistrar, but won't block on it.

Reply:
That makes some sense, done.